cmd/internal/cli/search.go

```diff
@@ -75,15 +74,18 @@ var SearchCmd = &cobra.Command{
 	DisableFlagsInUseLine: true,
 	Args:                  cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		searchFor := args[len(args)-1]
-		transport, _ := uri.Split(SearchRegistryURI)
-		switch transport {
```

Comment:
@DrDaveD this switch can still provide useful insight if incompatible transport

Reply:
I don't think it does provide anything useful in this function.  The option that fills in the SearchLibraryURI variable is called `--library`.  As it is, if somebody tries to put a transport on that for example by doing
```
apptainer search --library oras://ghcr.io/apptainer lolcow
```
It currently gives an error
```
FATAL:   Error initializing library client: unsupported protocol scheme "oras"
```

Reply:
I changed my mind, because I decided somebody might accidentally include a URI on the search query argument.  I added an error if any URI protocol scheme is provided there.